libs/vkd3d/resource.c

```diff
@@ -4432,6 +4432,25 @@ static void vkd3d_create_texture_srv(vkd3d_cpu_descriptor_va_t desc_va,
     if (key.u.texture.miplevel_count == VK_REMAINING_MIP_LEVELS)
         key.u.texture.miplevel_count = resource->desc.MipLevels - key.u.texture.miplevel_idx;
 
+    if ((vkd3d_config_flags & VKD3D_CONFIG_FLAG_PREALLOCATE_SRV_MIP_CLAMPS) &&
+        (desc->ViewDimension != D3D12_SRV_DIMENSION_TEXTURE2DMS) &&
+        (desc->ViewDimension != D3D12_SRV_DIMENSION_TEXTURE2DMSARRAY) &&
+        (hash_map_find(&resource->view_map.map, &key) == NULL))
+    {
+        uint32_t starting_mip = key.u.texture.miplevel_idx;
+        uint32_t mip_count = key.u.texture.miplevel_count != UINT32_MAX ?
+            key.u.texture.miplevel_count :
+            resource->desc.MipLevels - starting_mip;
```

Comment:
I think this is wrong. starting_mip is relative to the resource itself and we're iterating from starting_mip to MipLevels here.